pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/RealtimeKinesisIntegrationTest.java

```diff
@@ -0,0 +1,448 @@
+    addSchema(createKinesisSchema());
+    addTableConfig(createKinesisTableConfig());
+
+    createH2ConnectionAndTable();
+
+    // Push data into Kinesis
+    publishRecordsToKinesis();
+
+    // Wait for all documents loaded
+    waitForAllDocsLoadedKinesis(60_000L);
+  }
+
+  public Schema createKinesisSchema()
+      throws Exception {
+    URL resourceUrl = BaseClusterIntegrationTest.class.getClassLoader().getResource(SCHEMA_FILE_PATH);
+    Assert.assertNotNull(resourceUrl);
+    return Schema.fromFile(new File(resourceUrl.getFile()));
+  }
+
+  protected void waitForAllDocsLoadedKinesis(long timeoutMs)
+      throws Exception {
+    waitForAllDocsLoadedKinesis(timeoutMs, true);
+  }
+
+  protected void waitForAllDocsLoadedKinesis(long timeoutMs, boolean raiseError) {
+    TestUtils.waitForCondition(new Function<Void, Boolean>() {
+      @Nullable
+      @Override
+      public Boolean apply(@Nullable Void aVoid) {
+        try {
+          return getCurrentCountStarResult() >= totalRecordsPushedInStream;
+        } catch (Exception e) {
+          return null;
+        }
+      }
+    }, 1000L, timeoutMs, "Failed to load " + totalRecordsPushedInStream + " documents", raiseError);
+  }
+
+  public TableConfig createKinesisTableConfig() {
+    return new TableConfigBuilder(TableType.REALTIME).setTableName(getTableName()).setSchemaName(getTableName())
+        .setTimeColumnName("DaysSinceEpoch").setFieldConfigList(getFieldConfigs()).setNumReplicas(getNumReplicas())
+        .setSegmentVersion(getSegmentVersion()).setLoadMode(getLoadMode()).setTaskConfig(getTaskConfig())
+        .setBrokerTenant(getBrokerTenant()).setServerTenant(getServerTenant()).setIngestionConfig(getIngestionConfig())
+        .setLLC(true).setStreamConfigs(createKinesisStreamConfig()).setNullHandlingEnabled(getNullHandlingEnabled())
+        .build();
+  }
+
+  public Map<String, String> createKinesisStreamConfig() {
+    Map<String, String> streamConfigMap = new HashMap<>();
+    String streamType = "kinesis";
+    streamConfigMap.put(StreamConfigProperties.STREAM_TYPE, streamType);
+
+    streamConfigMap
+        .put(StreamConfigProperties.constructStreamProperty(STREAM_TYPE, StreamConfigProperties.STREAM_TOPIC_NAME),
+            STREAM_NAME);
+
+    streamConfigMap.put(
+        StreamConfigProperties.constructStreamProperty(STREAM_TYPE, StreamConfigProperties.STREAM_FETCH_TIMEOUT_MILLIS),
+        "30000");
+    streamConfigMap
+        .put(StreamConfigProperties.constructStreamProperty(STREAM_TYPE, StreamConfigProperties.STREAM_CONSUMER_TYPES),
+            StreamConfig.ConsumerType.LOWLEVEL.toString());
+    streamConfigMap.put(StreamConfigProperties
+            .constructStreamProperty(STREAM_TYPE, StreamConfigProperties.STREAM_CONSUMER_FACTORY_CLASS),
+        KinesisConsumerFactory.class.getName());
+    streamConfigMap
+        .put(StreamConfigProperties.constructStreamProperty(STREAM_TYPE, StreamConfigProperties.STREAM_DECODER_CLASS),
```

Comment:
lets make this use the new JsonDecoder from your other PR

Reply:
done